File: maximum-ascending-subarray-sum.ts
{
    function maxAscendingSum(nums: number[]): number {
        let c , r;
        r = 0;
        c = nums[0];

        for (let i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                c += nums[i]
            } else { 
                r = Math.max(c, r);
                c = nums[i];
            }
        }

        return Math.max(r, c);
    };

    console.log(maxAscendingSum([100,10,1]))
}

File: maximum-ascending-subarray-sum.rs
use std::cmp;

pub fn max_ascending_sum(nums: Vec<i32>) -> i32 {
    let mut c = nums[0];
    let mut r = 0;

    for i in 1..nums.len() {
        if (nums[i] > nums[i - 1]) {
            c += nums[i];
        } else {
            r = cmp::max(c, r);
            c = nums[i];
        }
    }

    cmp::max(c, r)
}